types/options.d.ts

```diff
@@ -39,6 +41,21 @@ export interface ComponentOptions {
   delimiters?: [string, string];
 }
 
+export interface FunctionalComponentOptions {
+  props?: string[] | { [key: string]: PropOptions | Constructor | Constructor[] };
+  functional: boolean;
+  render(this: never, createElement: $createElement, context: ContextObject): VNode;
+  name?: string;
+}
+
+export interface ContextObject {
```

Comment:
I think it would be better to rename this interface to more concrete name, like `RenderContext`?